### PR TITLE
NFC Update to Slang v3

### PR DESCRIPTION
**Pull request by fabianschuiki**

### CODE REVIEW

**Comment by albertethon** [comment]
slang 3.0 and 4.0 both use c++20

File: CMakeLists.txt [hunk 1]
```diff
@@ -521,26 +521,33 @@ if(CIRCT_SLANG_FRONTEND_ENABLED)
...
     FetchContent_Declare(
       slang
       GIT_REPOSITORY https://github.com/MikePopoloski/slang.git
-      GIT_TAG v2.0
+      GIT_TAG v3.0
       GIT_SHALLOW ON)
+    set(SLANG_INCLUDE_TOOLS OFF CACHE INTERNAL "")
+    set(SLANG_INCLUDE_TESTS OFF CACHE INTERNAL "")
+    set(SLANG_INCLUDE_INSTALL OFF CACHE INTERNAL "")
+    set(SLANG_USE_MIMALLOC OFF CACHE INTERNAL "")
```
Comment (albertethon) [inline CMakeLists.txt]:
```suggestion
    set(SLANG_USE_MIMALLOC OFF CACHE INTERNAL "")
    set(FETCHCONTENT_TRY_FIND_PACKAGE_MODE "NEVER")
```
I tried this in my machine and works well

File: CMakeLists.txt [hunk 2]
```diff
@@ -521,26 +521,33 @@ if(CIRCT_SLANG_FRONTEND_ENABLED)
...
     # harder than it ought to be.
-    set_property(GLOBAL APPEND PROPERTY CIRCT_EXPORTS slang_slang
-                                                      unordered_dense)
-    install(TARGETS slang_slang unordered_dense EXPORT CIRCTTargets)
-    set(CMAKE_CXX_FLAGS ${original_CXX_FLAGS})
+    set_property(GLOBAL APPEND PROPERTY CIRCT_EXPORTS
+      slang_slang unordered_dense fmt)
+    install(TARGETS slang_slang unordered_dense fmt EXPORT CIRCTTargets)
+    set(CMAKE_CXX_FLAGS ${original_CMAKE_CXX_FLAGS})
+    set(BUILD_SHARED_LIBS ${original_BUILD_SHARED_LIBS})
```
Comment (fabianschuiki) [inline CMakeLists.txt]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -521,26 +521,33 @@ if(CIRCT_SLANG_FRONTEND_ENABLED)
   if(CIRCT_SLANG_BUILD_FROM_SOURCE)
     # Build slang as part of CIRCT (see https://sv-lang.com/building.html)
     message(STATUS "Building slang from source")
-    set(original_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+    set(original_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+    set(original_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
     set(CMAKE_CXX_FLAGS "")
+    set(BUILD_SHARED_LIBS OFF)  # statically link slang into ImportVerilog
```
Comment (fabianschuiki) [inline CMakeLists.txt]:
```suggestion
    set(CMAKE_CXX_FLAGS "")
```

**Comment by fabianschuiki** [comment]
> both use c++20

Hmmm yeah good point. I've had more success building slang 3.0 on older compilers though :slightly_frowning_face:

File: CMakeLists.txt [hunk 4]
```diff
@@ -521,26 +521,31 @@ if(CIRCT_SLANG_FRONTEND_ENABLED)
...
+      GIT_TAG v3.0
       GIT_SHALLOW ON)
+    set(SLANG_INCLUDE_TOOLS OFF CACHE INTERNAL "")
+    set(SLANG_INCLUDE_TESTS OFF CACHE INTERNAL "")
+    set(SLANG_INCLUDE_INSTALL OFF CACHE INTERNAL "")
+    set(SLANG_USE_MIMALLOC OFF CACHE INTERNAL "")
+    set(FETCHCONTENT_TRY_FIND_PACKAGE_MODE "NEVER")
     FetchContent_MakeAvailable(slang)
     # The following feels *very* hacky, but CMake complains about the
     # CIRCTImportVerilog target linking against slang_slang (even with PRIVATE
```
Comment (fabianschuiki) [inline CMakeLists.txt:539]:
```suggestion
    # CIRCTImportVerilog target linking against slang_svlang (even with PRIVATE
```

**Pull request merged**